src/app/home/page.tsx

```diff
@@ -66,50 +46,142 @@ function HomePage() {
 		}
 	};
 
-	const fetchTxs = async () => {
+	const fetchPolicies = async () => {
 		const authToken = await getToken();
 		if (authToken && lockers) {
-			const lockersWithTxs = await getTokenTxs(authToken, lockers);
-			setLockers(lockersWithTxs);
+			setPolicies(await getPolicies(authToken, lockers[0].id as number));
 		}
 	};
 
-	const fetchPolicies = async () => {
+	// Handle locker updates from Supabase subscription
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	const handleLockerUpdates = async (payload: any) => {
+		console.log("Locker Change received!", payload);
 		const authToken = await getToken();
 		if (authToken && lockers) {
-			setPolicies(await getPolicies(authToken, lockers[0].id as number));
+			const updatedLockers = lockers.map((locker) =>
+				locker.id === payload.new.id
+					? { ...locker, ...payload.new }
+					: locker
+			);
+			setLockers(updatedLockers);
+			if (updatedLockers && updatedLockers.length > 0) {
+				const lockersWithTxs = await getTokenTxs(
+					authToken,
+					updatedLockers
+				);
+				setLockers(lockersWithTxs);
+				const policiesArray = await getPolicies(
+					authToken,
+					lockersWithTxs[0].id as number
+				);
+				setPolicies(policiesArray);
+			}
 		}
 	};
 
-	// Fetch lockers every 5 seconds if lockers is null
+	// Handle token transaction updates from Supabase subscription
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	const handleTxUpdates = async (payload: any) => {
+		console.log("Tx Change received!", payload);
+		// TODO: finish implementing this
```

Comment:
Looks unimplemented

Reply:
It looks like it is an async state issue. The `lockers` state variable is still null on the first render, so the if-statement that triggers tx subscription is not executing. I'll get this fixed.

Reply:
Should be fixed now

Reply:
Should be working now